### ISSUE TITLE
Error on fetching events in westend

### ISSUE THREAD

**Issue by TheGoldenEye** [polkadot-js/api#2663]
The following error occurs with @polkadot/api v2.0.1

```
Unable to decode storage system.events: createType(Vec<EventRecord>):: findMetaEvent: Unable to find Event with index 0x0801/[8,1]
[31m2020-09-29 20:48:32[39m [31m       RPC-CORE:[39m getStorage(key: StorageKey, at?: BlockHash): StorageData:: createType(Vec<EventRecord>):: findMetaEvent: Unable to find Event with index 0x0801/[8,1]
Error: createType(Vec<EventRecord>):: findMetaEvent: Unable to find Event with index 0x0801/[8,1]
    at createTypeUnsafe (/share/data/_jens/src/coins/polkadot/polka-store/polka-store/node_modules/@polkadot/types/create/createType.js:68:11)
    at Rpc._formatStorageData (/share/data/_jens/src/coins/polkadot/polka-store/polka-store/node_modules/@polkadot/rpc-core/index.js:421:40)
    at Rpc._formatOutput (/share/data/_jens/src/coins/polkadot/polka-store/polka-store/node_modules/@polkadot/rpc-core/index.js:387:21)
    at callWithRegistry (/share/data/_jens/src/coins/polkadot/polka-store/polka-store/node_modules/@polkadot/rpc-core/index.js:249:62)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

and here is the code:
```
const blocks: number[] = [2143450, 2143451];
  for (let i = 0, n = blocks.length; i < n; i++) {
    console.log(blocks[i]);
    const hash = await api.rpc.chain.getBlockHash(blocks[i]);
    await api.query.system.events.at(hash);
  }
```

the output:
```
polka-store: v1.0.2
Chain:       Westend
Node:        Parity Polkadot v0.8.24-5cbc418a-x86_64-linux-gnu
Provider:    ws://127.0.0.1:9944
API:         @polkadot/api v2.0.1

2143450
2143451
Unable to decode storage system.events: createType(Vec<EventRecord>):: findMetaEvent: Unable to find Event with index 0x0801/[8,1]
2020-09-29 20:51:03        RPC-CORE: getStorage(key: StorageKey, at?: BlockHash): StorageData:: createType(Vec<EventRecord>):: findMetaEvent: Unable to find Event with index 0x0801/[8,1]
```

btw, how can I catch these errors?
try/catch does not work...

**Comment by TheGoldenEye** [comment]
Hint: error occures only with node v0.8.24

**Comment by TheGoldenEye** [comment]
can you reproduce the bug?

**Comment by jacogr** [comment]
Not at all. 2143450 is innocent, 2143451 has payouts.

**Comment by TheGoldenEye** [comment]
I 've checked it again: The error occurs reproducibly at block 2143451. With node version v0.8.24-5cbc418a-x86_64-linux-gnu.
With and without flag `--wasm-execution`.
No error with older node version v0.8.23-d327000a-x86_64-linux-gnu

**Comment by TheGoldenEye** [comment]
other question: How can I catch these errors?
Is there something like .on('error', ...) ? I tried try/catch, but there is no exception to catch.
My goal is to catch the error and log it together with the block number.
Because the error message itself does not contain a block number, it is very difficult to find the exact location of the error.

**Comment by jacogr** [comment]
Not `--wasm-execution` (which swaps between interpreted and compiled modes), but rather `--execution wasm` (which says to always use the code from the chain, instead of "native when available").

**Comment by TheGoldenEye** [comment]
with `--execution wasm` the error does not occur

may I ask again:
How can I catch these 'RPC-CORE' errors?
Is there something like .on('error', ...) ? I tried try/catch, but there is no exception to catch.
My goal is to catch the error and log it together with the block number.
Because the error message itself does not contain a block number, it is very difficult to find the exact location of the error.

**Comment by jacogr** [comment]
Ok, so when it works with `--execution wasm` there is something seriously weird. Is that an actual `.24` release binary, or built from master? (There was a time when master said .24 and had the 24 runtime, but it was not compatible - the versions only get bumped at release)

On a normal `await` like in `await api.query.system.events.at(hash)` the error should be thrown. 

From a quick look at https://github.com/polkadot-js/api/blob/master/packages/api/src/promise/Api.ts#L61-L76 it may be that only the initial formatting error (on input being invalid) vs outputs failing is handled. (So there may be a gap there)

**Comment by TheGoldenEye** [comment]
my v0.8.24 is from the official deb disribution (which is same as the release binary: <https://github.com/paritytech/polkadot/releases/download/v0.8.24/polkadot> )

**Comment by jacogr** [comment]
I don’t have answers as to why the data comes back differently- actually since those blocks are in the 43 runtime version range, it will/should always use wasm execution anyway. 

Would be interesting to see on .25 (due for release tomorrow)

**Comment by TheGoldenEye** [comment]
ok, I will check this with .25, if released.

back to the error catching:
I've checked it again and the following code works as expected: (already with @polkadot/api **v2.0.1**!!!):
```
try {
    const hash = await api.rpc.chain.getBlockHash(2143451);
    await api.query.system.events.at(hash);
  }
  catch (e) {
    console.log('Exception:', (e as Error).message)
  }
```
Now I wonder why the error is not thrown in my code?
https://github.com/TheGoldenEye/polka-store/blob/master/src/utils.ts#L94-L99

fetchBlock is implemented here:
https://github.com/TheGoldenEye/polka-store/blob/master/src/ApiHandler.ts#L57
Can you please take a look at it? You would help me a lot. Thx

**Comment by TheGoldenEye** [comment]
the bug seems to be fixed in v0.8.25.
testet with @polkadot/api v2.1.1

**Comment by jacogr** [comment]
Closing.
